kotest-assertions/kotest-assertions-core/src/commonMain/kotlin/io/kotest/assertions/nondeterministic/eventually.kt

```diff
@@ -121,7 +121,7 @@ data class EventuallyConfiguration(
 }
 
 object EventuallyConfigurationDefaults {
-   var duration: Duration = Duration.INFINITE
+   var duration: Duration = INFINITE
```

Comment:
I am concerned over this approach in general. This thing is shared (because `object`) and mutable (because `var`). As such, there will be race conditions. We should refactor away the very need for shared mutable things. WDYT @OliverO2

Reply:
since this is default setting I think it is ok to change `var` to `val`

Reply:
@AlexCue987 I agree.

_All_ properties should really be `val` (though I wonder how this would affect backward compatibility in this case). Global state is almost never a good idea and in particular with the option of running parallel tests and concurrent coroutines, any context-induced configuration should be coroutine-local. See https://github.com/kotest/kotest/pull/2791 for an example.

I also wonder why that object is part of the public API. It might better be private.

Reply:
I think its appropriate to make another pr and deal with this issue
on this pr I will only focus on eventually's function
I will raise another pr for this issue!

Reply:
Yes, that makes sense. To make this PR ready for review, we'd need some test code covering the proposed fixes. Would also be good to update the initial PR comment with references to issues resolved by this PR.

Reply:
> I think its appropriate to make another pr and deal with this issue on this pr I will only focus on eventually's function I will raise another pr for this issue!

sure - this is a preexisting condition, not something introduced in this PR

Reply:
> Yes, that makes sense. To make this PR ready for review, we'd need some test code covering the proposed fixes. Would also be good to update the initial PR comment with references to issues resolved by this PR.

done with test!